docs/explanations/deprecation_policy.rst

```diff
@@ -0,0 +1,68 @@
+Ubuntu Pro Client Feature Deprecation Policy
+********************************************
+
+When pro-client introduces new features, particuarly features designed for
+other software to integrate with, we assume those features become relied-upon
+in all Ubuntu releases we support. That includes the oldest and also the newest
+(at time of writing, the oldest is 16.04 and the newest is 24.04).  The newest
+supported Ubuntu release typically has ~12 years of life left between LTS, ESM
+and Legacy coverage. That means we need to support any given feature in a
+backwards compatible way for at least 12 years.  This is particularly true for
+any integration point with machine-readable output (e.g. our API functions, any
+command with json formatted output). It is also true for important features
+that, even if lacking intentional machine-usability, are likely to be used in
+automation (e.g. any form of “pro attach” or “pro enable” that doesn't require
+interaction).
+
+Removing features by release
+============================
+
+We maintain the LTS promise of stability for all the LTS's we support,
+therefore we cannot remove a given feature from any LTS that already has the
+feature.  Instead, we will remove features starting on a devel Ubuntu release
+onward. To make the transition more apparent to consumers and give them time to
+adjust, we will only remove features in interim devel releases. We will not
+remove a feature for the first time in an LTS devel release. When the given
+feature is called on a release where it has been removed, pro-client will
+return an error code and message describing what to do instead of using the
+removed feature.
+
+This “per-release-removal” will be implemented with simple checks in the
```

Comment:
Do we need to mention this? Feels like an implementation detail that may not be relevant to users